lib/source-map-consumer.js

```diff
@@ -12,157 +12,153 @@ const base64VLQ = require("./base64-vlq"); // eslint-disable-line no-unused-vars
 const readWasm = require("../lib/read-wasm");
 const wasm = require("./wasm");
 
-function SourceMapConsumer(aSourceMap, aSourceMapURL) {
-  let sourceMap = aSourceMap;
-  if (typeof aSourceMap === "string") {
-    sourceMap = util.parseSourceMapInput(aSourceMap);
+const INTERNAL = Symbol("smcInternal");
+
+class SourceMapConsumer {
+  constructor(aSourceMap, aSourceMapURL) {
+    // If the constructor was called by super(), just return Promise<this>.
+    // Yes, this is a hack to retain the pre-existing API.
+    if (aSourceMap == INTERNAL) {
+      return Promise.resolve(this);
+    }
+
+    return _factory(aSourceMap, aSourceMapURL);
   }
 
-  return Promise.resolve().then(_ => {
-    return sourceMap.sections != null
-      ? new IndexedSourceMapConsumer(sourceMap, aSourceMapURL)
-      : new BasicSourceMapConsumer(sourceMap, aSourceMapURL);
-  });
-}
+  static initialize(opts) {
+    readWasm.initialize(opts["lib/mappings.wasm"]);
```

Comment:
This line isn't covered in the tests.  Does running the tests in the browser depend on #158 ?